### PR TITLE
Eslint update config to 1.5.1 compatibility

### PR DESCRIPTION
**Pull request by anselmbradford**
Update ESLint config for  version 1.5.1 compatibility.

## Additions

- Added additional environments and rules for version 1.5.1 ESLint compatibility:
  - Added `commonjs`, `jest`, `protractor` environments.
  - Added new ESLint `no-useless-concat`, `global-require`, `jsx-quotes`, `no-restricted-syntax`, `block-spacing`, `require-jsdoc`, `space-before-keywords`, `no-dupe-class-members`, `prefer-arrow-callback`, and `prefer-template` rules.
  - Added `properties` attribute of `id-length` rule.
  - Added `keywords`, `unnecessary`, and `numbers` attributes to `quote-props` rules.

## Review

- @ascott1  
- @wpears  
- @mistergone

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -30,11 +30,14 @@ parser: espree
 env:
   browser: true
   node: true
+  commonjs: false
```
Comment (ascott1) [inline .eslintrc]: Should this be set to `true` when using CommonJS modules?

**Comment by ascott1** [comment]
I have a question above, but otherwise :+1:

File: .eslintrc (hunk 1 above)
Comment (anselmbradford) [inline .eslintrc]: It only needs to be turned on if the linter complains about CommonJS syntax being missing (like `modules.exports`). This environments section turns on expected variables and the sort (which could also be manually added to the `globals` section). It's mostly useful for testing where there would be an eslintrc override file in the testing directory that turned on mocha, protractor, etc.

Comment (ascott1) [inline .eslintrc]: Makes sense to me!

**Comment by jimmynotjim** [comment]
:+1:

**Comment by Scotchester** [comment]
@anselmbradford Please update the grunt-eslint dependency version in generator-cf's `package.json` template accordingly.

**Comment by anselmbradford** [comment]
grunt-eslint update --> https://github.com/cfpb/generator-cf/pull/98

**Pull request merged**